packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2300,12 +2300,50 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
     }
   }
 
+  Set<SemanticsAction> _mutedSemanticsActions = <SemanticsAction>{};
+
+  /// Update the set of [SemanticsAction] that will be ignore when constructing
+  /// semantics action handlers.
+  ///
+  /// The muted [SemanticsAction] will apply to all the children in subtree.
+  void updateMutedSemanticsActions(Set<SemanticsAction> actions) {
+    assert(actions != null);
+    if (_mutedSemanticsActions.length == actions.length &&
+        _mutedSemanticsActions.every(actions.contains))
+      return;
+    _mutedSemanticsActions = actions;
+    markNeedsSemanticsUpdate();
+    bool markRenderObjectNeedsSemanticsUpdate(RenderObject renderChild) {
+      renderChild.markNeedsSemanticsUpdate();
+      renderChild.visitChildrenForSemantics(markRenderObjectNeedsSemanticsUpdate);
+    }
+    visitChildrenForSemantics(markRenderObjectNeedsSemanticsUpdate);
+  }
+
+  // Use [_inheritedMutedSemanticsActions] to access
+  Set<SemanticsAction> _cachedMutedSemanticsActions;
+
+  /// The muted [SemanticsAction] that aggregated from parent and current render
+  /// object.
+  Set<SemanticsAction> get _inheritedMutedSemanticsActions {
+    if (_cachedMutedSemanticsActions == null) {
+      _cachedMutedSemanticsActions = <SemanticsAction>{};
+      if (parent is RenderObject) {
+        final RenderObject renderParent = parent;
+        _cachedMutedSemanticsActions.addAll(renderParent._inheritedMutedSemanticsActions);
```

Comment:
How is it guaranteed that the parent's _inheritedMutedSemanticsActions are up-to-date at this point?

Reply:
if any ancestor has change, its subtree should have all their _cachedMutedSemanticsActions set to null